### PR TITLE
feat(broker): specify variables to include in workflow result

### LINKED ISSUE #3226: I can specify the set of variables to return in workflow outcome
**Is your feature request related to a problem? Please describe.**
By default when a worflow instance completes, all variables in the root scope are returned. The set of variables that should be returned can be specified in the requests.

**Describe the solution you'd like**
Include a list of variable names in the request.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* Added another field in grpc message 'CreateWorkflowInstanceWithResultRequest'
* Response will now include only the requested variables

## Related issues

closes #3226 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks @deepthidevaki can we discuss some things ?

@saig0 had the idea of removing the response record - for the gateway we do not it (since we do not write it to the stream) what do you think?

File: clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java [hunk 1]
```diff
@@ -111,5 +112,24 @@
...
 
   interface CreateWorkflowInstanceWithResultCommandStep1
-      extends FinalCommandStep<WorkflowInstanceResult> {}
+      extends FinalCommandStep<WorkflowInstanceResult> {
+
+    /**
+     * Set a list of variables names which should be fetched in the response.
+     *
+     * @param fetchVariables set of names of variables to be included in the response
+     * @return the builder for this command. Call {@link #send()} to complete the command and send *
```
Comment (Zelldon) [inline clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java]:
```suggestion
     * @return the builder for this command. Call {@link #send()} to complete the command and send
```

File: clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java [hunk 2]
```diff
@@ -111,5 +112,24 @@
...
+     *     it to the broker
+     */
+    CreateWorkflowInstanceWithResultCommandStep1 fetchVariables(List<String> fetchVariables);
+
+    /**
+     * Set a list of variables names which should be fetched in the response.
+     *
+     * @param fetchVariables set of names of variables to be included in the response
+     * @return the builder for this command. Call {@link #send()} to complete the command and send *
```
Comment (Zelldon) [inline clients/java/src/main/java/io/zeebe/client/api/command/CreateWorkflowInstanceCommandStep1.java]:
```suggestion
     * @return the builder for this command. Call {@link #send()} to complete the command and send
```

**Review by ChrisKujawa — APPROVED** [review]

**Pull request merged**